lib/datadog/appsec/utils/read_write_lock.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Datadog
+  module AppSec
+    module Utils
+      # Simple implentation for a fair read write lock
+      # It allows for as many concurrent readers but only one concurrent writer.
+      # To avoid the writer thread being exhausted, once a writer thread has signaled that it wants to acquire the lock,
+      # no more reader threads can acquire the lock.
+      # The new reader threads will wait for the writer thread to finish.
```

Comment:
I'm still a bit curious if this is the best policy for our situation; would it be better to perhaps have some kind of a softer limit where the writer tries to wait for a period without any readers for a while without "forcing itself" in, and after that period we fall back to the "writer gets priority" policy?